pkg/capi/gcp.go

```diff
@@ -0,0 +1,306 @@
+		Entry("SecureBoot and IntegrityMonitoring enabled", gcpv1.SecureBootPolicyEnabled, gcpv1.VirtualizedTrustedPlatformModulePolicyDisabled, gcpv1.IntegrityMonitoringPolicyEnabled),
+		Entry("all Shielded VM options disabled", gcpv1.SecureBootPolicyDisabled, gcpv1.VirtualizedTrustedPlatformModulePolicyDisabled, gcpv1.IntegrityMonitoringPolicyDisabled),
+		*/
+	)
+	// Define constants as variable for this case to pass values properly
+	confidentialComputeEnabled := gcpv1.ConfidentialComputePolicyEnabled
+	confidentialComputeDisabled := gcpv1.ConfidentialComputePolicyDisabled
+	DescribeTable("should configure Confidential VM correctly", framework.LabelCAPI, framework.LabelDisruptive,
+		func(confidentialCompute *gcpv1.ConfidentialComputePolicy) {
+			mapiProviderSpec := getGCPMAPIProviderSpec(cl)
+			Expect(mapiProviderSpec).ToNot(BeNil())
+			gcpMachineTemplate = createGCPMachineTemplate(mapiProviderSpec)
+			gcpMachineTemplate.Spec.Template.Spec.ConfidentialCompute = confidentialCompute
+			gcpMachineTemplate.Spec.Template.Spec.InstanceType = "n2d-standard-4"
+
+			if *confidentialCompute == gcpv1.ConfidentialComputePolicyEnabled {
+				mapiProviderSpec.OnHostMaintenance = OnHostMaintenanceTerminate
+				gcpMachineTemplate.Spec.Template.Spec.OnHostMaintenance = (*gcpv1.HostMaintenancePolicy)(&mapiProviderSpec.OnHostMaintenance)
+			} else {
+				mapiProviderSpec.OnHostMaintenance = "Migrate"
+				gcpMachineTemplate.Spec.Template.Spec.OnHostMaintenance = (*gcpv1.HostMaintenancePolicy)(&mapiProviderSpec.OnHostMaintenance)
+			}
+
+			Expect(cl.Create(ctx, gcpMachineTemplate)).To(Succeed())
+
+			By("Creating a MachineSet for Confidential VM")
+			machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, framework.NewCAPIMachineSetParams(
+				"gcp-machineset-confidential-74703",
+				clusterName,
+				mapiProviderSpec.Zone,
+				1,
+				corev1.ObjectReference{
+					Kind:       "GCPMachineTemplate",
+					APIVersion: infraAPIVersion,
+					Name:       gcpMachineTemplate.Name,
+				},
+			))
+			Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI MachineSet with Confidential VM configuration")
+
+			framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+
+			By("Verifying the Confidential VM configuration on the created GCP MachineTemplate")
+			createdTemplate := &gcpv1.GCPMachineTemplate{}
+			Expect(cl.Get(framework.GetContext(), client.ObjectKey{
+				Namespace: framework.ClusterAPINamespace,
+				Name:      gcpMachineTemplate.Name,
+			}, createdTemplate)).To(Succeed())
+
+			expectedConfidentialComputeValue := ptr.To(
+				map[bool]string{
+					true:  "Enabled",
+					false: "Disabled",
+				}[*confidentialCompute == gcpv1.ConfidentialComputePolicyEnabled],
+			)
+
+			Expect(fmt.Sprintf("%v", *expectedConfidentialComputeValue)).To(Equal(fmt.Sprintf("%v", *confidentialCompute)))
```

Comment:
I'm not really sure what/why we are testing this? This appears to just be checking the value of the input to the test? Should it not instead be comparing the value from the template itself?

We are creating the CAPI template, so why are we checking the values on the template?

Reply:
Earlier was using the template to compare missed it in latest commit , I will go through the code you suggested , will commit that 

```
var confidentialComputevalue = *createdTemplate.Spec.Template.Spec.ConfidentialCompute
			Expect(fmt.Sprintf("%v", confidentialComputevalue)).To(Equal(fmt.Sprintf("%v", *confidentialCompute)))
		}
```